// vite-plugin-fonts@0.2.2/dist/index.js
"use strict";Object.defineProperty(exports, "__esModule", {value: true});// src/google-fonts.ts
var GoogleFontsBase = "https://fonts.googleapis.com/css2";
function injectFonts({
  families,
  text,
  preconnect = true,
  display = "swap"
}) {
  const specs = [];
  const deferedSpecs = [];
  const tags = [];
  if (!Array.isArray(families)) {
    console.warn("Google font families is required");
    return tags;
  }
  if (families.length >= 0) {
    for (const family of families) {
      if (typeof family === "string") {
        deferedSpecs.push(family);
        continue;
      }
      if (!family)
        continue;
      const {
        name,
        styles,
        defer = true
      } = family;
      if (!name) {
        console.warn("A google font family name is missing");
        continue;
      }
      let spec = name;
      if (typeof styles === "string")
        spec += `:${styles}`;
      if (defer)
        deferedSpecs.push(spec);
      else
        specs.push(spec);
    }
  }
  if (preconnect && specs.length + deferedSpecs.length > 0) {
    tags.push({
      tag: "link",
      attrs: {
        rel: "preconnect",
        href: "https://fonts.gstatic.com/",
        crossorigin: true
      }
    });
  }
  if (deferedSpecs.length > 0) {
    let href = `${GoogleFontsBase}?family=${deferedSpecs.join("&family=")}`;
    if (typeof display === "string" && display !== "auto")
      href += `&display=${display}`;
    if (typeof text === "string" && text.length > 0)
      href += `&text=${text}`;
    tags.push({
      tag: "link",
      attrs: {
        rel: "preload",
        as: "style",
        onload: "this.rel='stylesheet'",
        href
      }
    });
  }
  if (specs.length > 0) {
    let href = `${GoogleFontsBase}?family=${specs.join("&family=")}`;
    if (typeof display === "string" && display !== "auto")
      href += `&display=${display}`;
    if (typeof text === "string" && text.length > 0)
      href += `&text=${text}`;
    tags.push({
      tag: "link",
      attrs: {
        rel: "stylesheet",
        href
      }
    });
  }
  return tags;
}
var google_fonts_default = injectFonts;

// src/typekit.ts
var TypekitFontBase = "https://use.typekit.net/";
function injectFonts2({
  id,
  defer = true
}) {
  const tags = [];
  if (typeof id !== "string") {
    console.warn("A Typekit id is required");
    return tags;
  }
  if (defer) {
    tags.push({
      tag: "link",
      attrs: {
        rel: "preload",
        as: "style",
        onload: "this.rel='stylesheet'",
        href: `${TypekitFontBase}${id}.css`
      }
    });
  } else {
    tags.push({
      tag: "link",
      attrs: {
        rel: "stylesheet",
        href: `${TypekitFontBase}${id}.css`
      }
    });
  }
  return tags;
}
var typekit_default = injectFonts2;

// src/index.ts
function VitePluginFonts(options = {}) {
  return {
    name: "vite-plugin-fonts",
    transformIndexHtml() {
      const tags = [];
      if (options.typekit)
        tags.push(...typekit_default(options.typekit));
      if (options.google)
        tags.push(...google_fonts_default(options.google));
      return tags;
    }
  };
}
var src_default = VitePluginFonts;



exports.Plugin = VitePluginFonts; exports.default = src_default;
